src/components/Species/SpeciesList.tsx

```diff
@@ -58,119 +60,112 @@ const SpeciesList = ({ species_list }: Props) => {
                         size="small"
                         startIcon={<Icon baseClassName="icons">search</Icon>}
                         onClick={() => {
-                            setShowAllSpecies(false);
                             if (searchFieldValue !== '') {
+                                setShowAllSpecies(false);
                                 setSearching(true);
                                 getData<SpeciesSummary[]>(
                                     `species/fuzzymatch/?query_str=${searchFieldValue}`,
                                     (data) => {
-                                        if (data.length === 0 && speciesSearchResults !== null) {
+                                        if (data.length === 0) {
                                             setSpeciesSearchResults(null);
                                         } else {
-                                            setSearching(false);
                                             setSpeciesSearchResults(data);
                                         }
+                                        setSearching(false);
```

Comment:
We should set `searching` to false on error too, which means we need to update `getData` to receive an onError callback. It can call the callback, if exists, after calling `console.error`.